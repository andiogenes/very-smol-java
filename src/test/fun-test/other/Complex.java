// Объявление класса
class Main {
    // Члены класса

    // Объявление класса
    class Nested {
        // Члены класса

        // Поле
        int foo;
        // Поле
        long bar;
        // Поле
        short baz;
        // Поле
        double foobar;

        // Метод
        void method1() {
            println(12345678);
        }

        // Метод
        int method2() {
            // Возврат из метода
            return 1;
        }
    }

    void bar() {
        println(987654321);
    }

    int param;

    int factorial() {
        switch (param <= 1) {
            case 1: return 1;
        }
        int term = param;
        param = param - 1;
        return term * factorial();
    }

    int intMethod() {
        return 9;
    }

    double doubleMethod() {
        return intMethod();
    }

    // Метод
    void main() {
        // Составной оператор
        {
            // Объявление переменной
            int foo;
            // Объявление переменной
            long bar;
            // Объявление переменной
            short baz;
            // Объявление переменной
            double foobar;
        }
        // Пустые операторы
        ;;;;;;;;;
        // Составные операторы
        {{{{}}}}
        // Определение переменной
        int foo = 5;
        // Определение переменной
        long bar = 25252534;
        // Определение переменной
        short baz = 15;
        // Определение переменной
        double foobar = 0.1E1;

        // Оператор switch
        switch (foo) {
            // Метка ветви switch
            case 1:
                // Тело ветви switch
            // Метка ветви switch
            case 2:
                // Тело ветви switch
                // Присваивание
                foo = 5;
            // Метка ветви switch
            case 3:
                // Тело ветви switch
                // Присваивание
                foo = 10;
            // Метка ветви switch
            case 4:
                // Тело ветви switch
                // Оператор break
                break;
            // Метка ветви switch
            case 5: /*оператор break*/
                println(42424242);
                break;
            default:
        }

        // Определение переменной
        int foobarbaz = /* унарный плюс */ +10;
        foo = foobarbaz = /* унарный минус */ -15;

        // Равенство
        short pseudoboolean1 = foo == foobarbaz;
        // Неравенство
        short pseudoboolean2 = foo != foobarbaz;
        // Логические И и ИЛИ
        short pseudoboolean3 = !(pseudoboolean1 || pseudoboolean2) || (pseudoboolean1 && !pseudoboolean2);

        // Операции сравнения
        short pb1 = 5 > 5;
        short pb2 = 7 < 8;
        short pb3 = 8 >= 6;
        short pb4 = 4 <= 4;

        // Арифметические операции (аддитивные и мультипликативные)
        foo = (5 + 5 - 9) * 45;
        // Инкремент (префиксный, постфиксный)
        foo++;
        ++foo;
        // Декремент (префиксный, постфиксный)
        --foo;
        foo--;
        // Деление
        foobar = foobar / 10;

        // Доступ по идентификатору (оператор-выражение);
        foobar;
        // Доступ к полю
        foo = Main.Nested.foo;
        // Вызов метода объекта
        Main.Nested.method1();
        // Вызов метода текущего объекта
        bar();

        // Пример работы рекурсии
        param = 7;
        int result = factorial(); // 7! = 5040

        {
            int foo = intMethod() / 2;
            double foobar = doubleMethod() / 2;
        }
    }
}