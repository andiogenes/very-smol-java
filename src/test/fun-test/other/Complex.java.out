foo : INT = 0 : INT
bar : LONG = 0 : LONG
baz : SHORT = 0 : SHORT
foobar : DOUBLE = 0.0 : DOUBLE
param : INT = 0 : INT
foo : INT = 0 : INT
bar : LONG = 0 : LONG
baz : SHORT = 0 : SHORT
foobar : DOUBLE = 0.0 : DOUBLE
foo : INT = 5 : INT
bar : LONG = 25252534 : LONG
baz : SHORT = 15 : SHORT
foobar : DOUBLE = 1.0 : DOUBLE
42424242
foobarbaz : INT = 10 : INT
foobarbaz : INT = -15 : INT
foo : INT = -15 : INT
pseudoboolean1 : SHORT = 1 : SHORT
pseudoboolean2 : SHORT = 0 : SHORT
pseudoboolean3 : SHORT = 1 : SHORT
pb1 : SHORT = 0 : SHORT
pb2 : SHORT = 1 : SHORT
pb3 : SHORT = 1 : SHORT
pb4 : SHORT = 1 : SHORT
foo : INT = 45 : INT
foobar : DOUBLE = 0.1 : DOUBLE
foo : INT = 0 : INT
12345678
987654321
param : INT = 7 : INT
term : INT = 7 : INT
param : INT = 6 : INT
term : INT = 6 : INT
param : INT = 5 : INT
term : INT = 5 : INT
param : INT = 4 : INT
term : INT = 4 : INT
param : INT = 3 : INT
term : INT = 3 : INT
param : INT = 2 : INT
term : INT = 2 : INT
param : INT = 1 : INT
result : INT = 5040 : INT
foo : INT = 4 : INT
foobar : DOUBLE = 4.5 : DOUBLE
